src/main/java/io/vertx/core/eventbus/impl/clustered/ClusteredEventBus.java

```diff
@@ -111,63 +104,50 @@ static void setTrustOptions(TCPSSLOptions sslOptions, TrustOptions options) {
   }
 
   @Override
-  public void start(Handler<AsyncResult<Void>> resultHandler) {
-    // Get the HA manager, it has been constructed but it's not yet initialized
-    HAManager haManager = vertx.haManager();
-    setClusterViewChangedHandler(haManager);
-    clusterManager.<String, ClusterNodeInfo>getAsyncMultiMap(SUBS_MAP_NAME, ar1 -> {
-      if (ar1.succeeded()) {
-        subs = ar1.result();
-        server = vertx.createNetServer(getServerOptions());
-
-        server.connectHandler(getServerHandler());
-        server.listen(asyncResult -> {
-          if (asyncResult.succeeded()) {
-            int serverPort = getClusterPublicPort(options, server.actualPort());
-            String serverHost = getClusterPublicHost(options);
-            serverID = new ServerID(serverPort, serverHost);
-            nodeInfo = new ClusterNodeInfo(clusterManager.getNodeID(), serverID);
-            vertx.executeBlocking(fut -> {
-              haManager.addDataToAHAInfo(SERVER_ID_HA_KEY, new JsonObject().put("host", serverID.host).put("port", serverID.port));
-              fut.complete();
-            }, false, ar2 -> {
-              if (ar2.succeeded()) {
-                started = true;
-                resultHandler.handle(Future.succeededFuture());
-              } else {
-                resultHandler.handle(Future.failedFuture(ar2.cause()));
-              }
-            });
+  public void start(Promise<Void> promise) {
+    server = vertx.createNetServer(getServerOptions());
+    server.connectHandler(getServerHandler());
+    server.listen(s -> {
+      if (s.succeeded()) {
```

Comment:
I think here we can benefit from flatmap for better readability

Reply:
Done